zerver/lib/bugdown/__init__.py

```diff
@@ -1169,6 +1170,19 @@ def __init__(self, parser: Any) -> None:
         super().__init__(parser)
         parser.markdown.tab_length = 4
 
+class EmoticonTranslationPreprocessor(markdown.preprocessors.Preprocessor):
+    """ Translates emoticons like `:)` into emoji like `:smile:`. """
+    def run(self, lines: List[Text]) -> List[Text]:
+        if db_data is None or not db_data['translate_emoticons']:
+            return lines
+
+        new_lines = []  # type: List[Text]
+
+        for line in lines:
+            new_lines.append(translate_emoticons(line))
```

Comment:
It seems like this will only translate syntax on a line by itself; is that the desired behavior?